eos-tech-support/eos-diagnostics

```diff
@@ -55,6 +55,217 @@ function collectBoardInfo() {
+    let fileEnum;
+    let output = '';
+
+    try {
+	fileEnum = dir.enumerate_children('standard::name',
+	                                  Gio.FileQueryInfoFlags.NONE, null);
+    } catch (e) {
+	return '';
+    }
+
+    /* This directory mixes i2c hosts and devices. Rather than try to
+     * parse them into a tree, we just have our loop iterations able to
+     * print useful info about both entry types. */
+
+    let info;
+    while ((info = fileEnum.next_file(null))) {
+	let devDir = dir.resolve_relative_path(info.get_name()).get_path();
+	output += info.get_name() + '\n';
+
+	output += '  name: ' + tryReadFile(devDir + '/name');
+
+        let contents = tryReadFile(devDir + '/firmware_node/path');
+	if (contents != '') {
+	    output += '  firmware_path: ' + contents;
+	}
+
+	try {
+	    let driver = GLib.file_read_link(devDir + '/device/driver');
+	    output += '  driver: ' + GLib.path_get_basename(driver) + '\n';
+	} catch (e) { }
+
+	try {
+	    let driver = GLib.file_read_link(devDir + '/driver');
+	    output += '  driver: ' + GLib.path_get_basename(driver) + '\n';
+	} catch (e) { }
+
+	output += '\n';
+    }
+
+    return output;
+}
+
+function collectHdaProcCodecInfo(dir) {
+    let output = '';
+    let fileEnum;
+
+    try {
+        fileEnum = dir.enumerate_children('standard::name',
+                                          Gio.FileQueryInfoFlags.NONE, null);
+    } catch (e) {
+        return '';
+    }
+
+    let info;
+    while ((info = fileEnum.next_file(null))) {
+        let name = info.get_name();
+        if (!name.startsWith("codec#")) {
+            continue;
+        }
+	output += dir.get_basename() + ' ' + name + ':\n';
+	output += tryReadFile(dir.resolve_relative_path(name).get_path());
+	output += '\n';
+    }
+
+    return output;
+}
+
+function collectHdaProcInfo() {
+    let dir = Gio.File.new_for_path("/proc/asound");
+    let fileEnum;
+    let output = '';
+
+    try {
+	fileEnum = dir.enumerate_children('standard::name',
+	                                  Gio.FileQueryInfoFlags.NONE, null);
+    } catch (e) {
+	return '';
+    }
+
+    let info;
+    while ((info = fileEnum.next_file(null))) {
+	let name = info.get_name();
+	if (name == "cards" || !name.startsWith("card")) {
+	    continue;
+	}
+	output += collectHdaProcCodecInfo(dir.resolve_relative_path(name));
+    }
+
+    return output;
+}
+
+function collectHdaSysInfo() {
+    let dir = Gio.File.new_for_path("/sys/class/sound");
+    let fileEnum;
+    let output = '';
+
+    try {
+	fileEnum = dir.enumerate_children('standard::name',
+	                                  Gio.FileQueryInfoFlags.NONE, null);
+    } catch (e) {
+	return '';
+    }
+
+    let info;
+    while ((info = fileEnum.next_file(null))) {
+	let dev = info.get_name();
+	let devDir = dir.resolve_relative_path(dev).get_path();
+
+	let initPins = tryReadFile(devDir + '/init_pin_configs').toString();
+	let driverPins = tryReadFile(devDir + '/driver_pin_configs').toString();
```

Comment:
No, when the file doesn't exist, tryReadFile returns an object of type object. I don't understand why.